diff -Naru python.ref/configure python/configure
--- python.ref/configure	2025-05-12 20:20:37.047821740 +0200
+++ python/configure	2025-05-12 20:22:19.978278039 +0200
@@ -15717,7 +15717,7 @@
 $as_echo "$ac_cv_func_memfd_create" >&6; }
   if test "x$ac_cv_func_memfd_create" = xyes; then :
 
-$as_echo "#define HAVE_MEMFD_CREATE 1" >>confdefs.h
+$as_echo "#define HAVE_MEMFD_CREATE_DISABLED 1" >>confdefs.h
 
 fi
 
diff -Naru python.ref/Makefile.pre.in python/Makefile.pre.in
--- python.ref/Makefile.pre.in	2025-05-12 20:20:36.865819165 +0200
+++ python/Makefile.pre.in	2025-05-12 20:21:39.418704190 +0200
@@ -498,6 +498,8 @@
 		$(OBJECT_OBJS) \
 		$(PYTHON_OBJS) \
 		$(MODULE_OBJS) \
+                $(LIBMPDEC_OBJS) \
+                $(LIBEXPAT_OBJS) \
 		$(MODOBJS)
 
 LIBRARY_OBJS=	\
